.github/workflows/build_garden.yaml

```diff
@@ -0,0 +1,182 @@
+name: C++ CI Workflow with conda-forge dependencies
+
+on:
+  push:
+    branches:
+      - master 
+  pull_request:
+    paths:
+      - collection-garden.yaml
+  schedule:
+  # * is a special character in YAML so you have to quote this string
+  # Execute a "nightly" build at 2 AM UTC
+  - cron:  '0 2 * * *' 
+
+env:
+  CACHE_NUMBER: 0
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            label: linux-64
+            conda_envs: /usr/share/miniconda/envs/gazebo-ws
+          - os: macos-latest
+            label: macos-64 
+            conda_envs: /usr/local/miniconda/envs/gazebo-ws
+          - os: windows-2019
+            label: win-64
+            conda_envs: C:\Miniconda\envs\gazebo-ws
+      fail-fast: false
+
+    name: '[${{ matrix.label }}@conda]'
+    runs-on: ${{ matrix.os }}
+    env:
+      # Workspace file to use for the build
+      VCS_WORKSPACE_FILE: collection-garden.yaml
+      COLCON_BUILD_ARGS: "--packages-up-to gz-common5" 
+      COLCON_TEST_ARGS: "--packages-up-to gz-common5" 
```

Comment:
I see you've added some dependencies above it though, like `ogre`.

It would be great to find a way to declare these dependencies in each package, a la `packages.apt`

Reply:
A possible idea is to have `environment.yaml` files for each package, and then merge them via something like https://github.com/amitbeka/conda-merge (even if I never used it). The tricky part is that we still need separate files for os-specific dependency, as `environment.win-64.yaml`, as you can't specify os-specific dependencies in a single `environment.yaml` file (see https://github.com/conda/conda/issues/8089).